File: frontend/src/lib/catalog.ts
import { apiGet } from "@/lib/api";

export interface CatalogSubcategory {
  slug: string;
  name: string;
  search_term: string;
}

export interface CatalogCategory {
  slug: string;
  name: string;
  description: string;
  icon: string;
  subcategories: CatalogSubcategory[];
}

export const fetchCatalogCategories = () => apiGet<CatalogCategory[]>("/catalog/categories");

File: backend/models/catalog.py
from pydantic import BaseModel, Field


class CatalogSubcategory(BaseModel):
    slug: str
    name: str
    search_term: str


class CatalogCategory(BaseModel):
    slug: str
    name: str
    description: str
    icon: str
    subcategories: list[CatalogSubcategory] = Field(default_factory=list)